lib/jnpr/jsnapy/jsnapy.py

```diff
@@ -312,19 +317,28 @@ def get_hosts(self):
         conf_file = self.args.file
         check = self.args.check
         snap = self.args.snap
-        if os.path.isfile(conf_file):
-            config_file = open(conf_file, 'r')
-            self.main_file = yaml.load(config_file)
-        elif os.path.isfile(os.path.join(get_path('DEFAULT', 'config_file_path'), conf_file)):
-            fpath = get_path('DEFAULT', 'config_file_path')
-            config_file = open(os.path.join(fpath, conf_file), 'r')
-            self.main_file = yaml.load(config_file)
+        if conf_file is not None:
+            if os.path.isfile(conf_file):
+                config_file = open(conf_file, 'r')
+                self.main_file = yaml.load(config_file)
+            elif os.path.isfile(os.path.join(get_path('DEFAULT', 'config_file_path'), conf_file)):
+                fpath = get_path('DEFAULT', 'config_file_path')
+                config_file = open(os.path.join(fpath, conf_file), 'r')
+                self.main_file = yaml.load(config_file)
         else:
-            self.logger.error(
-                colorama.Fore.RED +
-                "ERROR!! Config file '%s' is not present " %
-                conf_file, extra=self.log_detail)
-            sys.exit(1)
+            if self.args.hostname and self.args.login and self.args.passwd:
```

Comment:
self.args.login and self.args.passwd are optional, so you should check them.
Say if user has configured ssh, they dont need to pass password. Please remove that